src/AnimalClub/Skellygen/Skellygen.hs

```diff
@@ -168,17 +169,44 @@ _generatePotatoMesh p_snM44 p_thick skn = selfLocalMesh :| (mconcat cmeshes) whe
   absM44 = p_snM44 !*! relm44
   cmeshes = parMap rdeepseq (toList . _generatePotatoMesh absM44 thick) (_snChildren skn)
 
--- TODO switch to G.concat
+-- | same as _generatePotatoMesh except populates snDebugMesh
+_generatePotatoMeshWithDebugging ::
+  (AnimalFloat a)
+  => M44 a -- ^ parent ABS transform
+  -> a -- ^ parent thickness
+  -> SkellyNode a -- ^ node to generate
+  -> (SkellyNode a, NonEmpty (PotatoMesh a)) -- ^ output mesh and SkellyNode with debug mesh populated
+_generatePotatoMeshWithDebugging p_snM44 p_thick skn = (rSN, rMesh) where
+  thick = _snThickness skn
+  relm44 = _snM44Rel skn
+  selfLocalMesh = if _snIsPhantom skn
+    then emptyPotatoMesh
+    else transformPotatoMeshM44 p_snM44 $ generateSinglePotatoMesh relm44 thick p_thick
+  absM44 = p_snM44 !*! relm44
+  (children, cmeshes) = unzip $ parMap rdeepseq (over _2 toList . _generatePotatoMeshWithDebugging absM44 thick) (_snChildren skn)
+  rMesh = selfLocalMesh :| (mconcat cmeshes)
```

Comment:
Same `foldr` comment about maybe not needing laziness.

Reply:
I'm confused, I'm not doing any folds here.

Reply:
I looked up this because I didn't know what mconcat was https://hackage.haskell.org/package/base-4.9.1.0/docs/Data-Monoid.html `mconcat = foldr mappend mempty`

Reply:
I think you're using the default implementation but I didn't check >.>

Reply:
I see, that makes sense 👍